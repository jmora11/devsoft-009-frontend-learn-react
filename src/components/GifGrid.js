import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h1 className='animate__animated animate__bounce'>{category}</h1>

            { loading && <p>Loading...</p> }

            <div className='card-grid'>
                {
                    imagenes.map( img =>
                        <GifGridItem
                            key={ img.id }    
                            { ...img }
                        />  
                    )
                }
            </div>
        </>
    )
};
