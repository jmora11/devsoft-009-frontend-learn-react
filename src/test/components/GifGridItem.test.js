import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

const title = 'Demon Slayer';
const url = 'http://mock-test';

const img = {
    title,
    url
};

describe('Gif Grid Item', () => {

    let wrapper = shallow( <GifGridItem { ...img} /> );


    beforeEach( () => {
        wrapper = shallow( <GifGridItem { ...img} /> );
    });

    test('Debe mostrar el componente correctamente con props', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe tener la imagen bien la url y el alt', () => {
        const img = wrapper.find('img');
        expect ( img.prop('src') ).toBe( url );
        expect ( img.prop('alt') ).toBe( title );
    });

    test('debe tener la clase animate__fadeInBottomLeft', () => {
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeInBottomLeft')).toBe(true);
    });
});

