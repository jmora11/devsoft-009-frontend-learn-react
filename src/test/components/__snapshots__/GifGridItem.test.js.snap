// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid Item Debe mostrar el componente correctamente con props 1`] = `
<div
  className="card animate__animated animate__fadeInBottomLeft"
>
  <img
    alt="Demon Slayer"
    src="http://mock-test"
  />
</div>
`;
